docs/deploy-lab.md

```diff
@@ -0,0 +1,355 @@
+
+    ```yaml
+      build-deploy:
+        runs-on: ubuntu-latest
+
+        steps:
+          - uses: actions/checkout@v3
+
+          - name: Log in to Azure
+            uses: azure/login@v1
+            with:
+            creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+          - name: Build and deploy Container App
+            uses: azure/container-apps-deploy-action@v0
+            with:
+              appSourcePath: ${{ github.workspace }}/load_balancer/ # Location of Dockerfile
+              acrName: ${{ secrets.ACR_NAME }} # Name of Azure Container Registry
+              containerAppName: ${{ secrets.CONTAINER_APP_NAME }} # Name of Azure Container App
+              resourceGroup: ${{ secrets.RESOURCE_GROUP }} # Name of Azure Resource Group
+    ```
+
+    > **NOTE**: The `appSourcePath` is the location of the Dockerfile. The `acrName` is the name of the Azure Container Registry. The `containerAppName` is the name of the Azure Container App. The `resourceGroup` is the name of the Azure Resource Group.
+
+7. Add the following to define the `test-deployment` job. This job will determine the staging URL of the newly deployed revision and use a GitHub Action to ping the API endpoint to ensure it is responding. If the health check succeeds, the traffic manager on the container app will be updated to point all traffic at the newly deployed container.
+
+    ```yaml
+      test-deployment:
+        needs: build-deploy
+        runs-on: ubuntu-latest
+
+        steps:
+        - name: Log in to Azure
+          uses: azure/login@v1
+          with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+        - name: Get new container name
+          run: |
+            # Enable extension installation
+            az config set extension.use_dynamic_install=yes_without_prompt
+            # Get the last deployed revision name
+            REVISION_NAME=`az containerapp revision list -n ${{ secrets.CONTAINER_APP_NAME }} -g ${{ secrets.RESOURCE_GROUP }} --query "[].name" -o tsv | tail -1`
+            # Get the last deployed revision's fqdn
+            REVISION_FQDN=`az containerapp revision show -n ${{ secrets.CONTAINER_APP_NAME }} -g ${{ secrets.RESOURCE_GROUP }} --revision "$REVISION_NAME" --query properties.fqdn -o tsv`
+            # Store values in env vars
+            echo "REVISION_NAME=$REVISION_NAME" >> $GITHUB_ENV
+            echo "REVISION_FQDN=$REVISION_FQDN" >> $GITHUB_ENV
+        - name: Test deployment
+          id: test-deployment
+          uses: jtalk/url-health-check-action@v3 # Marketplace action to touch the endpoint
+          with:
+            url: "https://${{ env.REVISION_FQDN }}/api" # Staging endpoint
+        - name: Deploy succeeded
+          run: |
+            echo "Deployment succeeded! Enabling new revision"
+            az containerapp ingress traffic set -n ${{ secrets.CONTAINER_APP_NAME }} -g ${{ secrets.RESOURCE_GROUP }} --revision-weight "${{ env.REVISION_NAME }}=100"
```

Comment:
We just launched [NGINX for Azure](https://azuremarketplace.microsoft.com/en-us/marketplace/apps/f5-networks.f5-nginx-for-azure?tab=Overview&exp=ubp8) but I'm not sure if it:
1. Supports the free tier we are working in for this lab
2. Supports specifically this kind of functionality (I think it would likely be accomplished by applying a new NGINX config to the ingress).

To be clear, I think there's no change here - but I wonder if you'd be open to using that if the billing works and our team could work out the commands (it's new to us as well)?

Reply:
Replied via email